File: fix.py
import re

with open('src/App.jsx', 'r', encoding='utf-8') as f:
    content = f.read()

# Replace React import
if 'Suspense' not in content:
    content = re.sub(r'import React from ["\']react["\'];', 'import React, { Suspense } from "react";', content)

# Regex to find imports
import_regex = re.compile(r'import\s+(\w+)\s+from\s+["\']\./(pages|components)/(.*?)["\'];')

lazy_imports = []

def replace_func(match):
    comp_name = match.group(1)
    folder = match.group(2)
    file_name = match.group(3)
    if comp_name in ['Navbar', 'Footer', 'ShopBanner', 'Collection']:
        return match.group(0)
    
    lazy_imports.append(f'const {comp_name} = React.lazy(() => import("./{folder}/{file_name}"));')
    return ''

content = import_regex.sub(replace_func, content)

# Add lazy imports
content = content.replace('// 🛡️ Route Guards', '\n'.join(lazy_imports) + '\n\n// 🛡️ Route Guards')

if '<Suspense' not in content:
    content = content.replace('<Routes>', '<Suspense fallback={<div style={{height: "100vh", display: "flex", justifyContent: "center", alignItems: "center"}}>Loading...</div>}>\n        <Routes>')
    content = content.replace('</Routes>', '</Routes>\n        </Suspense>')

with open('src/App.jsx', 'w', encoding='utf-8') as f:
    f.write(content)

print("App.jsx updated with lazy loading.")

File: fix.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'src', 'App.jsx');
let content = fs.readFileSync(filePath, 'utf-8');

// Replace standard react import with Suspense
if (!content.includes('Suspense')) {
    content = content.replace(/import React from ["']react["'];/, 'import React, { Suspense } from "react";');
}

// Find all imports from pages and components, but NOT Navbar, Footer, ShopBanner
const importRegex = /import (\w+) from ["']\.\/(pages|components)\/(.+?)["'];/g;
const lazyImports = [];

content = content.replace(importRegex, (match, componentName, folder, file) => {
  if (['Navbar', 'Footer', 'ShopBanner', 'Collection'].includes(componentName)) {
    return match; // Keep these static
  }
  lazyImports.push('const ' + componentName + ' = React.lazy(() => import("./' + folder + '/' + file + '"));');
  return ''; // Remove the static import
});

// Add lazy imports
content = content.replace(/\/\/ \uD83D\uDEE1\uFE0F Route Guards/g, lazyImports.join('\n') + '\n\n// \uD83D\uDEE1\uFE0F Route Guards');

// Wrap Routes in Suspense
if (!content.includes('<Suspense fallback')) {
    content = content.replace(/<Routes>/g, '<Suspense fallback={<div style={{height: "100vh", display: "flex", justifyContent: "center", alignItems: "center"}}>Loading...</div>}>\n        <Routes>');
    content = content.replace(/<\/Routes>/g, '</Routes>\n        </Suspense>');
}

fs.writeFileSync(filePath, content);
console.log('App.jsx optimized with React.lazy');
